core/code/utils_misc.js

```diff
@@ -474,6 +474,42 @@ window.clampLatLngBounds = function (bounds) {
   return L.latLngBounds(clampLatLng(SW), clampLatLng(NE));
 }
 
+
+/*
+pnpoly Copyright (c) 1970-2003, Wm. Randolph Franklin
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+     disclaimers.
+  2. Redistributions in binary form must reproduce the above copyright notice in the documentation and/or other
+     materials provided with the distribution.
+  3. The name of W. Randolph Franklin may not be used to endorse or promote products derived from this Software without
+     specific prior written permission.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+window.pnpoly = function(latlngs, point) {
```

Comment:
This is terribly wrong, as we cannot use latlngs here, this algorithm is for 2d (projected) points.

Reply:
that's from fix-china-offset. 
The usage of the algorithm in IITC was never correct in the first place since everything is about a geodesic, that pnpoly cannot cover, whatever the projection.
Question is, is this algorithm good enough for the usage we have in IITC.

Reply:
On a second aspect, we do want a function that works on latlng, more than a pnpoly.

Reply:
So let's add another,  more high-level function, operating by latlngs. Projecting them on 2d space, passing to pnpoly, and returning results,

Reply:
That would still be incorrect as we use geodesic instead of straight lines in draw tools.
But again, this is good enough and would match a tool that would not use geodesic (as any selection tool).
Maybe we need to extend drawtool (or use another plugin) to draw standard polyline/polygon.

Reply:
Or we can use the interpolated coordinates when using geodesic polygon as filtering

Reply:
What is interpolated coordinates?

Reply:
all points calculated to draw a geodesic polyline

Reply:
In fact there is not so much difference between geodesic and planar on those zoom levels where filtering makes sense

Reply:
that's also true with latlng vs projected

Reply:
Btw, draw items are geodesic, so filtering will be geodesic as well.

Reply:
my point, but what do you mean by "will be" ? is it "must/should/may be" ?

Reply:
I mean as we are using geodesic shapes for bounds - our calculations are geodesically correct. Right now.

Reply:
But why don't you use `_rings`?

Reply:
it's not because currently, we only use endpoints. Interpolated points are only used for rendering, `getLatLngs()` ignore them

Reply:
>  Interpolated points are only used for rendering

So why not make use of them? Why should we care about getLatLngs?